app/controllers/bucket_jobs_controller.rb

```diff
@@ -3,16 +3,17 @@ class BucketJobsController < ApplicationController
   before_action :set_job, only: [:update, :destroy]
 
   def create
-    job = Job.find(params[:job_id])
-    @bucket.add_job(job.id)
+    @job = Job.find(params[:job_id])
+    @bucket.add_job(@job.id)
     session[:bucket] = @bucket.contents
-    flash[:success] = "You have started an application for #{job.title}. You have started #{pluralize(@bucket.total_jobs, 'job')} applications."
-    redirect_to bucket_path(job) 
+    flash[:success] = "You have started an application for #{@job.title}. You have started #{pluralize(@bucket.total_jobs, 'job')} applications."
+    create_job_application
```

Comment:
@roscalabrin There's a good amount of logic that is starting to live here. I'd recommend splitting it out into a PORO. If you're not sure how, go ahead and ship this and create a story to refactor. We can sit maybe include  anyone that is interested as I live refactor 🙀